python/src/equistore/operations/reduce.py

```diff
@@ -0,0 +1,99 @@
+import numpy as np
+
+
+from equistore import Labels, TensorBlock, TensorMap
+
+from . import _dispatch
+
+# TODO maybe add a way to reduce only the rows
+#  with sample[sample_name] = idx
+# for example not summing over all the sample
+# that have the same 'structure'
+# but only those with 'structure' = 0
+
+
+def _reduction_sample_block(block: TensorBlock, sample_name: list) -> TensorBlock:
+    """Create a new :py:class:`TensorBlock` summing the `properties` among
+    the selected `samples`.
+    The output :py:class:`TensorBlocks` have the same components of the input one.
+    :param block: :py:class:`TensorBlock` -> input block
+    :param sample_name: list -> names of samples to sum
```

Comment:
you do not need to put the types here again, sphinx should be able to extract them from the signature

Reply:
this function does not show up in the doc. only resolve that acts on TensorMap does. I can put also this public

Reply:
No, I think this is fine being private. My comment was more about making docstrings consistent in the whole codebase